pkgs/development/interpreters/ruby/default.nix

```diff
@@ -145,6 +150,13 @@ let
           # Remove unnecessary groff reference from runtime closure, since it's big
           sed -i '/NROFF/d' $out/lib/ruby/*/*/rbconfig.rb
 
+          ${lib.optionalString removeReferenceToCC ''
+              # Get rid of the CC runtime dependency
+              ${removeReferencesTo}/bin/remove-references-to \
+                -t ${stdenv.cc} \
+                $out/lib/libruby.so.*
+          ''}
```

Comment:
could this be done every time? sounds interesting to remove it if we can

Reply:
Does this break jit compiling?

Reply:
> $ nix-shell -p ruby --run 'ruby -e "puts RbConfig::CONFIG[\'configure_args\']"'
> '--prefix=/nix/store/b3lhi1wlia02vyd94jk56bm3kvgf1yc0-ruby-2.6.5' '--bindir=/nix/store/b3lhi1wlia02vyd94jk56bm3kvgf1yc0-ruby-2.6.5/bin' '--sbindir=/nix/store/b3lhi1wlia02vyd94jk56bm3kvgf1yc0-ruby-2.6.5/sbin' '--includedir=/nix/store/b3lhi1wlia02vyd94jk56bm3kvgf1yc0-ruby-2.6.5/include' '--oldincludedir=/nix/store/b3lhi1wlia02vyd94jk56bm3kvgf1yc0-ruby-2.6.5/include' '--mandir=/nix/store/b3lhi1wlia02vyd94jk56bm3kvgf1yc0-ruby-2.6.5/share/man' '--infodir=/nix/store/b3lhi1wlia02vyd94jk56bm3kvgf1yc0-ruby-2.6.5/share/info' '--docdir=/nix/store/b3lhi1wlia02vyd94jk56bm3kvgf1yc0-ruby-2.6.5/share/doc/' '--libdir=/nix/store/b3lhi1wlia02vyd94jk56bm3kvgf1yc0-ruby-2.6.5/lib' '--libexecdir=/nix/store/b3lhi1wlia02vyd94jk56bm3kvgf1yc0-ruby-2.6.5/libexec' '--localedir=/nix/store/b3lhi1wlia02vyd94jk56bm3kvgf1yc0-ruby-2.6.5/share/locale' '--enable-shared' '--enable-pthread'   'CC=gcc' 'CXX=g++'

Dunno which gems need it at runtime, I wouldn’t risk it by default.

Reply:
Any gem that does need this at runtime probably wouldn’t work on a pure Nix system anyway, so I’d be happy disabling this across the board.

Reply:
Won’t do it in this PR, but feel free to open another one for the proper `ruby` package. :)
Though of course it would break systems which might be using that.